_data-prepper/pipelines/configuration/sinks/s3.md

```diff
@@ -59,33 +93,34 @@ Option | Required | Type | Description
 `region` | No | String | The AWS Region to use for credentials. Defaults to [standard SDK behavior to determine the Region](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/region-selection.html).
 `sts_role_arn` | No | String | The AWS Security Token Service (AWS STS) role to assume for requests to Amazon SQS and Amazon S3. Defaults to `null`, which will use the [standard SDK behavior for credentials](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials.html).
 `sts_header_overrides` | No | Map | A map of header overrides that the IAM role assumes for the sink plugin.
-`sts_external_id` | No | String | The external ID to attach to AssumeRole requests from AWS STS.
+`sts_external_id` | No | String | An STS external ID to use when Data Prepper assumes the role. For more information, see the documentation for `ExternalId` in the [STS AssumeRole](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html) reference.
+
 
 
 ## Threshold configuration
 
-Use the following options to set ingestion thresholds for the `s3` sink.
+Use the following options to set ingestion thresholds for the `s3` sink. When any of these conditions is met, Data Prepper will write events to an S3 object.
 
 Option | Required | Type | Description
 :--- | :--- | :--- | :---
-`event_count` | Yes | Integer | The maximum number of events the S3 bucket can ingest. 
-`maximum_size` | Yes | String | The maximum number of bytes that the S3 bucket can ingest after compression. Defaults to `50mb`.
-`event_collect_timeout` | Yes | String | Sets the time period during which events are collected before ingestion. All values are strings that represent duration, either an ISO_8601 notation string, such as `PT20.345S`, or a simple notation, such as `60s` or `1500ms`.
+`event_count` | Yes | Integer | The number of Data Prepper events to accumulate before writing an object to S3.
+`maximum_size` | No | String | The maximum number of bytes to accumulate before writing an object to S3. Defaults to `50mb`.
+`event_collect_timeout` | Yes | String | The maximum timeout before Data Prepper writes an event to S3. The value should be either an ISO-8601 duration, such as `PT2M30S`, or a simple notation, such as `60s` or `1500ms`.
```

Comment:
Something like "The maximum amount of time allowed before"?

Reply:
If we wanted to still use the word, "timeout", could we do, "The maximum duration before Data Prepper timesout when writing an event to S3"?

Reply:
That would change the technical meaning wouldn't it @dlvenable? Is "timeout duration" the length of time that Data Prepper has to write an event to S3 before timing out, or is the "timeout duration" the length of the time we set for Data Prepper to prepare to write an event? If its the second one, I feel like we should go @natebower original suggestion, since that technically doesn't fit the definition of "timeout" despite the word being used in the setting.

Reply:
@natebower , The term "timeout" is not really ideal either in the configuration or in the description. Nothing is really "timing out" here. This is really more of a flush interval. 

> or is the "timeout duration" the length of the time we set for Data Prepper to prepare to write an event? If its the second one, I feel like we should go @natebower original suggestion, since that technically doesn't fit the definition of "timeout" despite the word being used in the setting.

Yes, it is the second one. The amount of time before Data Prepper writes the event.

Perhaps we can say "The maximum amount of time before".

Reply:
Let's go with the "The maximum amount of time before."